### PR TITLE
Initial refactor

### PR DESCRIPTION
**Pull request by sosthene-nitrokey**
- Fix formatting and clippy warnings
- Update dependencies
- Add vsmartcard example
- Add tests file for easy tests
- Fix compatibility with [pivy-tool](https://github.com/arekinath/pivy)
- Remove request command chaining, which is handled by [apdu_dispatch](https://github.com/solokeys/apdu-dispatch/blob/main/src/dispatch.rs#L155)
- Use full names for cryptosystems

### CODE REVIEW

File: examples/virtual.rs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+// Now you should be able to see the card in `pcsc_scan` and talk to it with `piv-tool`r
+//
+// Set `RUST_LOG=piv_authenticator::card=info` to see the executed commands.
+
+// TODO: add CLI
+
+fn main() {
+    env_logger::init();
+
+    trussed::virt::with_ram_client("opcard", |client| {
```
Comment (robin-nitrokey) [inline examples/virtual.rs:17]: s/opcard/piv-authenticator/

File: Cargo.toml [hunk 2]
```diff
@@ -2,37 +2,53 @@
 name = "piv-authenticator"
 version = "0.0.0-unreleased"
 authors = ["Nicolas Stalder <n@stalder.io>"]
```
Comment (robin-nitrokey) [inline Cargo.toml:4]: add Nitrokey

**Comment by sosthene-nitrokey** [comment]
- Fix formatting and clippy warnings
- Update dependencies
- Add vsmartcard example
- Add tests file for easy tests
- Fix compatibility with [pivy-tool](https://github.com/arekinath/pivy)
- Remove request command chaining, which is handled by [apdu_dispatch](https://github.com/solokeys/apdu-dispatch/blob/main/src/dispatch.rs#L155)
- Use full names for cryptosystems
- Refactor state to properly handle failures to load the persistent state returns an error rather than overwriting it and still allows for factory reset for repeated failures.
- Add a CI

**Comment by sosthene-nitrokey** [comment]
There's also the issue of Licensing, I took some code from opcard and kept the initial LGPL License

**Comment by sosthene-nitrokey** [comment]
Yes, but do we also relicense the code that I took from opcard to MIT + Apache or do we make all new contributions be LGPL?

**Comment by sosthene-nitrokey** [comment]
I think @jans23 was in favor of having it under MIT+Apache like the original repo. Can you confirm @jans23 ?

**Comment by jans23** [comment]
I have a small preference for LGPL. Unless we have good reasons for another license (incl. complexity) I would prefer LGPL.

**Comment by sosthene-nitrokey** [comment]
Normally everything is right with [622fb5a](https://github.com/Nitrokey/piv-authenticator/pull/2/commits/622fb5a41d6c1e40c017053eed825d8ac252189a)

**Comment by robin-nitrokey** [comment]
Can you add a readme that mentions the original repository, author and licenses?

**Comment by sosthene-nitrokey** [comment]
@robin-nitrokey can I merge this?

**Comment by robin-nitrokey** [comment]
> @robin-nitrokey can I merge this?

From my point of view yes.  My only concern is how to integrate back into upstream but I leave that to your discretion.

**Pull request merged**